src/server/ApiRoutes/Hold.js

```diff
@@ -402,44 +402,19 @@ function createHoldRequestServer(req, res, pickedUpBibId = '', pickedUpItemId =
   );
 }
 
-function createHoldRequestEdd(req, res) {
-  // Ensure user is logged in
-  const loggedIn = User.requireUser(req);
-  if (!loggedIn) return false;
-
-  return postHoldAPI(
-    req,
-    req.body.itemId,
-    req.body.pickupLocation,
-    req.body.form,
-    req.body.itemSource,
-    (response) => {
-      const data = JSON.parse(response).data;
-      res.json({
-        id: data.id,
-        jobId: data.jobId,
-        pickupLocation: data.pickupLocation,
-      });
-    },
-    (error) => {
-      logger.error(
-        `Error calling postHoldAPI in createHoldRequestEdd, itemId: ${req.body.itemId}`,
-        error,
-      );
-      res.json({
-        status: error.status,
-        error,
-      });
-    },
-  );
-}
-
 function eddServer(req, res) {
   const {
     bibId,
     itemId,
     searchKeywords,
+    serverRedirect,
   } = req.body;
+
+  let { fromUrl } = req.body;
+  fromUrl = fromUrl ? `&fromUrl=${fromUrl}` : '';
+
+  res.respond = serverRedirect === false ? res.redirect : res.json;
```

Comment:
I think you want to have `res.redirect` when `serverRedirect` is true. So `serverRedirect ? res.redirect : res.json;`

Reply:
Thanks for catching this. Yes it looks like those conditions should be flipped so `serverRedirect === false ? res.json : res.redirect`. My only concern with `serverRedirect ? res.redirect : res.json` is I want redirecting to be the default behavior, so `res.json` is only if `serverRedirect` is explicity false, as opposed to just falsey.